### PR TITLE
Overhaul job metrics.

### PR DESCRIPTION
**Pull request by jmchilton**
See individual commits for atomic changes.
- Annotate some job metrics as being safe out of the box - no need to set ``expose_potentially_sensitive_job_metrics`` for these. (Implements #13294)
- Drop collectl support - project seems very dead, cgroups seems a lot more robust.
- Add a bunch of new unit tests for new and old functionality. 
- Ad a bunch of additional type hints throughout the module.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

**Comment by bgruening** [comment]
LGTM, restarted the one failing test.

**Comment by jmchilton** [comment]
The CI bug has been fixed in dev - this won't break the build because of that. Would people prefer I rebase though?

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
One of those guys thinks that Johns time is better spent doing other cool stuff than rebasing again :)

File: lib/galaxy/managers/jobs.py [hunk 1]
```diff
@@ -658,24 +662,21 @@ def summarize_job_metrics(trans, job):
...
-        title, value = trans.app.job_metrics.format(metric_plugin, metric_name, metric_value)
-        return dict(
-            title=title,
-            value=value,
-            plugin=metric_plugin,
-            name=metric_name,
-            raw_value=str(metric_value),
+    safety_level = Safety.SAFE
+    if trans.user_is_admin:
+        safety_level = Safety.UNSAFE
```
Comment (nsoranzo) [inline lib/galaxy/managers/jobs.py:667]: @jmchilton Shouldn't this be:
```python
    safety_level = Safety.UNSAFE
    if trans.user_is_admin:
        safety_level = Safety.SAFE
```

**Pull request merged**